generation/WinSDK/enums.json

```
@@ -34723,6 +34723,144 @@
       }
     ]
   },
+  {
+    "type": "uint",
+    "name": "DXGI_PRESENT",
+    "flags": true,
+    "autoPopulate": {
+      "filter": "DXGI_PRESENT_",
+      "header": "dxgi.h"
+    },
+    "members": [],
+    "uses": [
+      {
+        "interface": "IDXGISwapChain",
+        "method": "Present",
+        "parameter": "Flags"
+      },
+      {
+        "interface": "IDXGISwapChain1",
+        "method": "Present1",
+        "parameter": "PresentFlags"
+      },
+      {
+        "interface": "IDXGIDecodeSwapChain",
+        "method": "PresentBuffer",
+        "parameter": "Flags"
+      }
+    ]
+  },
+  {
+    "type": "uint",
+    "name": "DXGI_ENUM_MODES",
+    "flags": true,
+    "autoPopulate": {
+      "filter": "DXGI_ENUM_MODES_",
+      "header": "dxgi.h;dxgi1_2.h"
+    },
+    "members": [],
+    "uses": [
+      {
+        "interface": "IDXGIOutput",
+        "method": "GetDisplayModeList",
+        "parameter": "Flags"
+      },
+      {
+        "interface": "IDXGIOutput1",
+        "method": "GetDisplayModeList1",
+        "parameter": "Flags"
+      }
+    ]
+  },
+  {
+    "type": "uint",
+    "name": "DXGI_MWA",
+    "flags": true,
+    "autoPopulate": {
+      "filter": "DXGI_MWA_",
+      "header": "dxgi.h"
+    },
+    "members": [],
+    "uses": [
+      {
+        "interface": "IDXGIFactory",
+        "method": "MakeWindowAssociation",
+        "parameter": "Flags"
+      }
+    ]
+  },
+  {
+    "type": "uint",
+    "name": "DXGI_MAP",
+    "flags": true,
+    "autoPopulate": {
+      "filter": "DXGI_MAP_",
+      "header": "dxgi.h"
+    },
+    "members": [],
+    "uses": [
+      {
+        "interface": "IDXGISurface",
+        "method": "Map",
+        "parameter": "MapFlags"
+      }
+    ]
+  },
+  {
+    "type": "uint",
+    "name": "DXGI_RESOURCE_PRIORITY",
+    "flags": true,
+    "autoPopulate": {
+      "filter": "DXGI_RESOURCE_PRIORITY_",
+      "header": "dxgi.h"
+    },
+    "members": [],
+    "uses": [
+      {
+        "interface": "IDXGIResource",
+        "method": "SetEvictionPriority",
+        "parameter": "EvictionPriority"
+      },
+      {
+        "interface": "IDXGIResource",
+        "method": "GetEvictionPriority",
+        "parameter": "pEvictionPriority"
+      }
+    ]
+  },
+  {
+    "namespace": "Windows.Win32.Graphics.Dxgi",
+    "type": "uint",
+    "name": "DXGI_SHARED_RESOURCE_RW",
```

```suggestion
    "name": "DXGI_SHARED_RESOURCE_RIGHTS",
```

"Flags" also acceptable.

---

The type is named RW: https://learn.microsoft.com/en-gb/windows/win32/direct3ddxgi/dxgi-shared-resource-rw

Doesn't that confuse users who might try to `ctrl+F` this? And how about the documentation attribute: https://github.com/Microsoft/win32metadata/blob/09cc32d989ccb945458bb059742435f48aa0879d/generation/WinSDK/documentationMappings.rsp#L13835